MailManager/src/main/java/inc/softserve/resources/EnvelopeResource.java

```diff
@@ -21,25 +22,28 @@
 @Produces(MediaType.APPLICATION_JSON)
 public class EnvelopeResource {
 
-    private EnvelopeDao dao;
+    private EnvelopeDao envelopeDao;
+    private UserDao userDao;
     private EnvelopeTools envelopeTools;
 
     @Inject
-    public EnvelopeResource(EnvelopeDao dao, EnvelopeTools envelopeTools){
-        this.dao = dao;
+    public EnvelopeResource(EnvelopeDao envelopeDao, UserDao userDao, EnvelopeTools envelopeTools){
+        this.envelopeDao = envelopeDao;
+        this.userDao = userDao;
         this.envelopeTools = envelopeTools;
     }
 
     @POST
     @Timed
-    public long saveEnvelope(@QueryParam("Subject") String subject,
+    public String saveEnvelope(@QueryParam("Subject") String subject,
                                @QueryParam("To") String to,
                                @QueryParam("From") String username,
                                @QueryParam("template") String template,
                                @Context HttpServletRequest request){
-        Envelope envelope = new Envelope(EmailServiceApplication.users.get(username),
+        Envelope envelope = new Envelope(userDao.getUserByUsername(username),
                                             to, subject, template, EnvelopeState.IN_PROGRESS);
-        dao.saveOrUpdate(envelope);
+        String id = envelopeDao.saveOrUpdate(envelope);
+        envelope.setId(id);

```

Comment:
It should be better to let DAO explicitly return saves object and use something like `envelope = envelopeDao.saveOrUpdate(envelope);` rather then setting some fields (like Id) manually.